src/contracts/deploymentHelper/DeploymentHelper.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import "../vendored/interfaces/MultiTokenMediatorInterface.sol";
+import "../CowProtocolVirtualToken.sol";
+
+/// @dev Contract contains the logic for deploying CowProtocolVirtualToken on gnosis chain
+/// @author CoW Protocol Developers
+contract DeploymentHelper {
+    address public immutable foreignToken;
+    address public immutable multiTokenMediator;
+    bytes32 public immutable merkleRoot;
+    address payable public communityFundsTarget;
+    address public immutable gnoToken;
+    uint256 public immutable gnoPrice;
+    address public immutable wrappedNativeToken;
+    uint256 public immutable nativeTokenPrice;
+
+    constructor(
+        address _foreignToken,
+        address _multiTokenMediator,
+        bytes32 _merkleRoot,
+        address payable _communityFundsTarget,
+        address _gnoToken,
+        uint256 _gnoPrice,
+        address _wrappedNativeToken,
+        uint256 _nativeTokenPrice
+    ) {
+        foreignToken = _foreignToken;
+        multiTokenMediator = _multiTokenMediator;
+        merkleRoot = _merkleRoot;
+        communityFundsTarget = _communityFundsTarget;
+        gnoToken = _gnoToken;
+        gnoPrice = _gnoPrice;
+        wrappedNativeToken = _wrappedNativeToken;
+        nativeTokenPrice = _nativeTokenPrice;
+    }
+
+    function deploy() external returns (address) {
+        address bridgedToken = MultiTokenMediatorInterface(multiTokenMediator)
+            .bridgedTokenAddress(foreignToken);
```

Comment:
We will only know the address of the mainnet COW token shortly deployment time. This means this contract cannot be deployed too much in advance. Maybe it's better to have this as a parameter of the function to give the DAO more time to deploy in advance before the proposal?

Reply:
> This means this contract cannot be deployed too much in advance. 
Probably the last thing to determine for the cowToken is the merkleRoot, right? And this one will also be the last missing ingredient for this BridgedTokenDeployer. Hence, I think we can anyway not deploy it much in advance, can we?

Reply:
Good point, so we can't deploy this in advance unfortunately.